### PR TITLE
Circular alias error

### PR DESCRIPTION
**Pull request by paulvanbrenk**
this fixes a stack overflow when resolving aliases on imported modules.

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -574,7 +574,8 @@ module ts {
             if (node.moduleReference.kind === SyntaxKind.ExternalModuleReference) {
                 return resolveExternalModuleSymbol(resolveExternalModuleName(node, getExternalModuleImportEqualsDeclarationExpression(node)));
             }
-            return getSymbolOfPartOfRightHandSideOfImportEquals(<EntityName>node.moduleReference, node);
+            let symbol = getSymbolOfPartOfRightHandSideOfImportEquals(<EntityName>node.moduleReference, node);
+            return symbol && symbol.flags & SymbolFlags.Alias ? resolveAlias(symbol) : symbol;
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: Can you parenthesize the conditional, or break this into an `if`?

Comment (JsonFreeman) [inline src/compiler/checker.ts]: Please comment why this is necessary.

**Comment by JsonFreeman** [comment]
:+1:

**Comment by JsonFreeman** [comment]
Thinking about this more, I think the following code may be negatively affected by your fix:
```ts
module outer {
    export module mod { }
    export import alias = mod;
    export var alias: number;
}
import im = outer.alias; // Try to get the type of 'im'
```
I haven't tried this out. But if I'm not mistaken, before your fix, the type of `im` would be computed from the type of the var `alias`, which is number. Whereas now, it would obtain the `alias` symbol, see that it's an alias, and continue on to the module `mod`, which does not have a type.

**Comment by JsonFreeman** [comment]
In that case, I think the best fix will involve some additional plumbing meaning parameters.

**Pull request closed without merging**